lib/Sema/DebuggerTestingTransform.cpp

```diff
@@ -152,11 +159,17 @@ class DebuggerTestingTransform : public ASTWalker {
   /// recursively transform the children of the transformed expression, and 2)
   /// the transformed expression itself.
   std::pair<bool, Expr *> insertCheckExpect(Expr *OriginalExpr, Expr *DstExpr) {
-    auto *DstDRE = extractDecl(DstExpr);
-    if (!DstDRE)
+    auto *DstRef = extractDeclOrMemberRef(DstExpr);
+    if (!DstRef)
       return {true, OriginalExpr};
 
-    ValueDecl *DstDecl = DstDRE->getDecl();
+    ValueDecl *DstDecl;
+    if (auto *DRE = dyn_cast<DeclRefExpr>(DstRef))
+      DstDecl = DRE->getDecl();
+    else {
+      auto *MRE = cast<MemberRefExpr>(DstRef);
```

Comment:
should we assert here that this can be either a `MemberRefExpr` or a `DeclRefExpr` ?

Reply:
I'd expect the forced cast to MemberRefExpr makes this explicit.